exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc

```diff
@@ -200,29 +200,53 @@ void parse(std::string&& code, options_type& result) {
 
 } //namespace ParserInternals
 
+struct LinePositions {
+    size_t mStartPos;
+    size_t mEndPos;
+};
+
+inline std::optional<const LinePositions> getNextLine(const size_t current_pos, const std::string & scriptCode) {
+    /**
+     * Find first of occurence of '%', starting search from position 'current_pos'.
+     * If no '%' is found, return an empty result.
+     * If '%' is found, search backwards from '%' for '\n' or \r':
+     *  1. If not found, '%' was found in the first line. Then we can set 'new_option_start_pos'=0
+     *  2. If found, set new_option_start_pos to position 1 char behind pos of found '\n' or '\r'.
+     * Then search forward for next occurence of '\n' or \r' and assign to var 'line_end_pos':
+        1. If not found, 'line_end_pos' will get assigned std::string::npos (std::string::substr(...,npos), returns substring until end of string
+        2. If found, 'line_end_pos' will assigned to position of line end of line where '%' was found
+     */
+    std::optional<LinePositions> retVal;
+    const size_t new_option_start_pos = scriptCode.find_first_of("%", current_pos);
+    if (new_option_start_pos == std::string::npos)
+        return retVal;
+    size_t line_start_pos = scriptCode.find_last_of("\r\n", new_option_start_pos);
+    if (std::string::npos == line_start_pos)
+        line_start_pos = 0;
+    else
+        line_start_pos++;
+
```

Comment:
Not creating a scope via `{}` for **loops** and **if** statements is a **_very dangerous pattern_** which should be avoided!

```suggestion
    if (new_option_start_pos == std::string::npos) {
        return retVal;
    }
    size_t line_start_pos = scriptCode.find_last_of("\r\n", new_option_start_pos);
    if (std::string::npos == line_start_pos) {
        line_start_pos = 0;
    }
    else {
        line_start_pos++;
    }
```

Reply:
Done